### ISSUE TITLE
Atelier Ryza 2: Lost Legends & the Secret Fairy (1257290)

### ISSUE THREAD

**Issue by wwmm** [ValveSoftware/Proton#4569]
# Compatibility Report
- Name of the game with compatibility issues: Atelier Ryza 2: Lost Legends & the Secret Fairy
- Steam AppID of the game: 1257290

## System Information
- GPU: rx 5700 xt sapphire pulse
- Driver/LLVM version: Mesa 21.0.0-rc2
- Kernel version: 5.10.10
- Link to full system information report as Gist: [system info](https://gist.github.com/wwmm/28380aac1076eb49b25319727c2214bc)
- Proton: 1611589137 experimental-5.13-20210125
- SteamLinuxRuntime: v0.20210114.1-0-g5a1cbb7
- pressure-vessel: 0.20210114.0+srt1 scout 0.20210114.2
- soldier: 0.20210114.2 soldier 0.20210114.2
- Desktop: Gnome 3.38.2 on Wayland

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

Proton log:

[steam-1257290.zip](https://github.com/ValveSoftware/Proton/files/5876128/steam-1257290.zip)

The issue is so unusual to me that is hard to describe. By default this game starts in window mode. I noticed that the animation in the start menu was at 40 fps but felt a lot slower slower than that. But I was able to start the game. And the first impression was good. Solid 60 fps and no issue until Ryza got to the capital after the first battle. From this point on the game became unplayable. What is weird is that the framerate feels like 5 fps although Steam fps counter shows no less than 40 fps. Depending on where the camera is pointing it counts 60 fps. I don't know if heavy stuttering is the right word here. I am not a native English speaker. But it is almost like many frames are missing between one movement and the other.

Alt-tab is broken. The game window just stays over everything even if changing virtual desktops. But as strange as it may sound we can do alt-tab while the game shows the usual loading screen propaganda. It is only after getting to its start menu that things get weird. Like this image shows:

![Screenshot from 2021-01-26 18-09-00](https://user-images.githubusercontent.com/16225172/105907592-ebabb800-6003-11eb-9644-3ec65dc1da42.png)

The game window is detached from another that stays in the background.

I tried Proton 5.13-5 but there was no difference. I also tried switching between amdvlk and radv but nothing changed. Another attempt was using Xorg instead of Wayland. But things get worse. Instead of a slow start menu there is a white screen.

**Comment by wwmm** [comment]
> Instead of a slow start menu there is a white screen.

On Wayland a black screen with white border lines at the right and bottom side is shown instead of the start menu. But an attempt to alt-tab makes the screen visible. On Xorg this trick does not work and the white screen stays. But ironically I can alt-tab as expected on Xorg.

**Comment by wwmm** [comment]
I did a test after setting `VK_INSTANCE_LAYERS=VK_LAYER_KHRONOS_validation` and it showed lots of these messages in journalctl
```
jan 26 22:14:17 wwmm steam.desktop[78985]:     Objects: 1
jan 26 22:14:17 wwmm steam.desktop[78985]:         [0] 0x8d9f6000008d9f6, type: 23, name: NULL
jan 26 22:14:17 wwmm steam.desktop[78985]: VUID-vkCmdDrawIndexed-None-02699(ERROR / SPEC): msgNum: -1539028524 - Validation Error: [ VUID-vkCmdDrawIndexed-None-02699 ] Object 0: handle = 0x8d9f7000008d9f7, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xa44449d4 | VkDescriptorSet 0x8d9f7000008d9f7[] encountered the following validation error at vkCmdDrawIndexed() time: Descriptor in binding #25 index 0 requires FLOAT component type, but bound descriptor format is VK_FORMAT_R32_UINT. The Vulkan spec states: Descriptors in each bound descriptor set, specified via vkCmdBindDescriptorSets, must be valid if they are statically used by the VkPipeline bound to the pipeline bind point used by this command (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkCmdDrawIndexed-None-02699)
```

I have no idea if they are useful...

**Comment by wwmm** [comment]
In one of the game threads in the Steam forum someone directed a Linux user to this dxvk commit https://github.com/doitsujin/dxvk/commit/a17ec95f214499862b7ded21c64cb4d8f346aa68. I did a test now with `d3d9.deferSurfaceCreation = True` and it fixes the issues I reported. At some point I tried many dxvk options but I did not consider the possibility that a dx11 game would be using dx9 calls... Strange...

**Comment by wwmm** [comment]
Sometimes when loading another area the game freezes. Looking the at cpu usage I can see this leads to once core at 100%. So far this seems random. I have been trying to change a few settings but so far nothing seems to have an effect over it.

**Comment by wwmm** [comment]
Yesterday the game received a big update > 8 GB and now it does not start anymore. It freezes in the initialization phase with a screen that sometimes is black sometimes is white. And one of the CPU cores is stuck at 100 %. This is the PROTON log
[steam-1257290.log](https://github.com/ValveSoftware/Proton/files/5913089/steam-1257290.log)

**Comment by wwmm** [comment]
I wonder if it is the latest Soldier beta update that is causing this. Cyberpunk is also not starting anymore

**Comment by wwmm** [comment]
Ok. The latest Soldier Beta that is breaking everything...

**Comment by kisak-valve** [comment]
Hello @wwmm, please open a new issue report in the [steam-runtime](https://github.com/ValveSoftware/steam-runtime/issues/new?template=ISSUE_TEMPLATE.md) issue tracker with your findings.

**Comment by wwmm** [comment]
Done

**Comment by wwmm** [comment]
I was finally able to get a proton log with one of the random freezes I mentioned above. Not the one about the Soldier. The one I have seen in this game since the beginning. This time it happened while the game was initializing but it can happen when loading any of the game maps. And it always leads to a 100% cpu core usage. File attached
[steam-1257290.zip](https://github.com/ValveSoftware/Proton/files/5920364/steam-1257290.zip)

**Comment by wwmm** [comment]
In my logs I see `warn:  D3D11CoreCreateDevice: Adapter is not a DXVK adapter`. Is this a harmless warning or a sign something is broken?

**Comment by kisak-valve** [comment]
Hello @wwmm, my understanding is that log message means that Wine's DXGI is being used instead of DXVK's DXGI. This means that some DXVK tweaks are not available, but we need Wine's DXGI for VKD3D integration. Generally harmless on its own.

**Comment by wwmm** [comment]
After setting `PROTON_NO_ESYNC=1` I was able to play for many hours yesterday without any of the freezes when loading a map. It seems this game does not play nice with esync. Fortunately the performance drop is not that noticeable. I think it is around 3 to 5 fps.

**Comment by ffrasisti** [comment]
## System Information

Steam Deck. 512gb
Latest stable SteamOS

Proton version: 7.0.6/Experimental

## Symptoms
Game can hang on loading screens transitioning between areas. 
Pre rendered video cutscenes can either dont play or dont have audio

## Reproduce
Open the map on a savefile that has a few locations and jump between different areas. eventually, the loading screen will be infinite

Log. Im using PROTON_NO_ESYNC=1 and WINEDLLOVERRIDES="d3d11=n,b" %command% with https://github.com/doitsujin/atelier-sync-fix 
TOOK ME A WHILE THIS TIME BUT IT DID END UP IN THE INFINITE LOADING SCREEN AND I HAD TO QUIT FROM OVERLAY
Log is big, but im guessing what you need is at the end, since thats when it finally hanged
I though esync would help and i think it may have made the issue take longer to manifest. It's the same or worse without it. 7.0.6 exhibits the same behaviour and so does GE
[steam-1257290.zip](https://github.com/ValveSoftware/Proton/files/11130922/steam-1257290.zip)

**Comment by ffrasisti** [comment]
Upated with log. Problem seems to be the same as #6050 / #5841
